### PR TITLE
Fractal cache improvements

### PR DESCRIPTION
**Pull request by epistemancer**
Updates fractal cache:
* when the query has IN or IS filters on its split attributes (which limits the number of rows that can be returned), and it already has all the necessary rows in the cache, it will serve the result from the cache instead of requerying the database
* when the only thing missing from the cache is a uniqueCount apply, just count the rows being returned instead of querying the database just for that count

### CODE REVIEW

File: src/query/filter.ts [hunk 1]
```diff
@@ -511,6 +522,11 @@ module FacetLegacy {
     public toHash(): string {
       return "IS:" + this.attribute + ":" + this.value;
     }
+
+    public getCardinality(splits: FacetSplit[]): number {
+      var splitAttributes: string[] = splits.map(function (s) { return s.attribute; });
+      return (splitAttributes.indexOf(this.attribute) >= 0) ? 1 : null;
```
Comment (aprilrd) [inline src/query/filter.ts]: should it be `(splitAttributes.indexOf(this.attribute) >= 0) ? 1 : 0;`?

Comment (aprilrd) [inline src/query/filter.ts]: oops. nvm. Read it wrong.

Comment (epistemancer) [inline src/query/filter.ts]: No, it's actually supposed to be null

**Comment by epistemancer** [comment]
updated to change behavior of `AndFilter.getCardinality`

**Comment by aprilrd** [comment]
:stars:

**Pull request merged**